### ISSUE TITLE
Events stop after some time

### ISSUE THREAD

**Issue by ciscozine** [maragelis/ParadoxRs232toMqtt#27]
Hi, 
I have a Magellan 5000 and I use a WemosD1 mini to connect to the alarm system.

At night I arm the alarm, but in the morning "ParadoxRs232toMqtt" seems to be down.
I can ping the wemos but I cannot receive MQTT Message.

What can I do? How can I do to debug it?

I use hassio on a rasperry and "Mosquitto broker" as a MQTT Server.

thanks
Fabio

**Comment by maragelis** [comment]
have you tried the new release ? There where many bug fixes done.

**Comment by ciscozine** [comment]
Yes, I have tried the last release without resolve the problem

Thanks

**Comment by maragelis** [comment]
Is there any chance the raspberry is going to sleep?
Are you also getting "paradox disconnected" on the status topic ?

**Comment by ciscozine** [comment]
No It doesn't go to sleep.
Do you know where can I find MQTT broker log on hassio?

thanks

**Comment by maragelis** [comment]
No sorry. Use mqtt.fx and leave it connected to status topic

**Comment by ciscozine** [comment]
Ok. I'll update you. Thanks

**Comment by maragelis** [comment]
No I dont use home assistant. I use openhab and homekit

**Comment by maragelis** [comment]
If you want you can try this https://github.com/maragelis/ParadoxRs232toMqtt/tree/Dev/ParadoxAlarmSystem/ParadoxAlarmSystemOTA

It has some mqtt connection updates, and get back to me Thanks

**Comment by ciscozine** [comment]
What are the differences between the 2ino files?
thanks

**Comment by maragelis** [comment]
Yes it will reconnect

**Comment by ciscozine** [comment]
Hi, I'll load the new sketch. I'll update you tomorrow.
A question. I have trace =1 but using console (9600 or 115200) I do not see nothing. Why?

thanks

**Comment by ciscozine** [comment]
Again, the same problem. The interface to the paradox seems be down.

I have sniffed the IP flow.

The wemos is live (icmp ok), the mqtt flow still persist. I notice only a SSDP port unreachable  ssdp:

each five minute the wemos send to the broker a ssdp message, but at 3:06am the wemos send 2 ssdp message and 1 of that it's a port unreachable (problem on hassio server?)
![image](https://user-images.githubusercontent.com/46693480/52257352-ca36fa80-291a-11e9-894e-a8ff7f7ae1d9.png)
But I don't know if it is the problem.

2 questions:
1. My circuit has no resistor. is this the cause the problem?
2. Can you tell me for each lib, which version have you used? I updated some lib and now, if i try to compile the sketch, I see "DynamicJsonBuffer' was not declared in this scope"

Thanks again

**Comment by ciscozine** [comment]
I'll test and I'll update you! Thansk.
My personal suggestion for .bin file. Why don't you give use the possibility to enable authentication and SSL encryption? It would be perfect :)

Thanks

**Comment by maragelis** [comment]
I will look into it

**Comment by ciscozine** [comment]
I update you.
also with the .bin the problem still persist.
What I notice is that after the problem the log of my mqtt broker on hassio (raspberry) is:
1549393009: Client 80:7D:3A:3E:B5:FF has exceeded timeout, disconnecting.
1549393009: Socket error on client 80:7D:3A:3E:B5:FF, disconnecting.
but I can ping the wemos d1..

Moreover if I unplug the power and plug the power, the wemos d1 led blinks only 1 time and the ping stop to work.
If I detach the serial port to the alarm system and unplug/plug the power, the wemos d1 led blink correctly (3 or 4 times); then if I attach the serial port the broker receive the message correctly.

Thanks

**Comment by ciscozine** [comment]
One time I receive also a "{"status":"Problem connecting to panel"}".
I change the cable without any changes. This is why I ask you if it is better use some resistor .

thanks

**Comment by maragelis** [comment]
So this happens only when armed ? please help so I can help.

**Comment by ciscozine** [comment]
No I noticed that only when disarmed. I cannot test if armed because there is a PIR sensor that I cannot disable

**Comment by maragelis** [comment]
So you are disarmed and the wemos disconnects from mqtt ?

**Comment by ciscozine** [comment]
I'm sorry. I understand now. Generally when I try do disarm the system (it is armed), the wemos stops to work.

**Comment by maragelis** [comment]
Are you maybe having low wifi signal

**Comment by maragelis** [comment]
If the wifi signal cut out for too long the times out and cannot reconnect. Please check if you are getting disconnect status message.

**Comment by maragelis** [comment]
{"status":"Paradox Disconnected"} this is the last will message if wifi is disconnected, can you check if you are getting this message.

**Comment by ciscozine** [comment]
Well, I have an access point at 20cm .. It isn't a wireless problem; in fact i can ping the wemos without any problem.
No I don't see "{"status":"Paradox Disconnected"}" .

**Comment by ciscozine** [comment]
Paradox Magellan 5000. I don't know the firmware. Can I find it with your system?

**Comment by maragelis** [comment]
IF that is the one with the 433mhz transmitter make sure your cables and wemos are well grounded and shielded. I will look into it as I only have the SP series api ans get back to you. @[rjduraocosta](/rjduraocosta) are you using the same panel ?

**Comment by maragelis** [comment]
Could you if you have the means to read the voltage over your tx and rx lines. if they are above 3.3v I suggest a level converter between wemos and panel .

**Comment by maragelis** [comment]
Are you powering the wemos from the panel ?

**Comment by ciscozine** [comment]
I used power from the panel and from a external power supply. Each one have the problem.
tomorrow I check the cable and the welding.

thanks again

**Comment by rjduraocosta** [comment]
Hello @maragelis and @ciscozine ,,

I measured the voltage on my Wemos Pin's and they are the following:

- D8(TX) - 3,26 V
- D7(RX) - 4,85 V
The above makes sence because the Wemos transmits at 3.3V logic at the D8 pin and receives from the panel at D7 pin in 5V logic.

Some ideas:
Move the RX/TX to the original ones? 
Level Shifter? 
Resistors? 
RF Interference?

@ciscozine any news about your setup?

Thanks for the help.

**Comment by maragelis** [comment]
I am betting on interference. I will try tackle it in code but it could yield unwanted results.

**Comment by ciscozine** [comment]
Hi @maragelis , good news.

I have tested all components with the tester and it seems ok, but I have identified the cause of my problem (I hope). A solder. 

Thanks for all.

For the last but not the least, my suggestion is to make a bin file with the possibility to enable authentication and encryption (ssl).

**Comment by rjduraocosta** [comment]
Hello again,

I could try and move the Wemos further away for testing, but I cloud only do this at the weekend. It is all in a box that I have to move further and also do another longer cable.

Thanks for the suggestion.

**Comment by rjduraocosta** [comment]
Hello @maragelis good evening,

I have been testing the system and I found one thing, that it is the following:
- I have connected MQTT.fx monitoring the paradoxdCTL/# topic. After some time I see that I stop receiving the zone events while I walk around the house. After this I send a command (stay/disarm/etc.) to the system that the system takes well and I start receiving back the zone events. I have been repeting this for 2 hours and the system does not stay completly down if I keep "waiking" it up from now and then.

If you have some other idea other than mving the wemos away until this weekend, or something I could test I appreciate it.

I do not have a level converter I just have resistors right now. I am not sure if using resistors at the Wemos RX line is fast enough for the serial communication with this configuration: 
![image](https://user-images.githubusercontent.com/40459138/52371017-4da43900-2a4c-11e9-856b-38cfb075bdd9.png)

Thanks again for your work.

**Comment by maragelis** [comment]
ok so I am starting to understand what is going on. Can you please tell me the last message you receive before it goes down. Is it a mqtt disconnect ? 

The only way we will see whats going on is for you to use a ftdi adapter connect to d4 / RX and get Trace messages

**Comment by rjduraocosta** [comment]
I have never seen an mqtt disconnect message before the system goes down. In all the tests the last message in the mqtt broker was a zone event, sometimes ON event sometimes OFF event.

Tomorrow I will use an ftdi adapter and connect it to D4 to see the messages. Before that I have to enable TRACE=1through mqtt, right? Another thing, can I use the Arduino IDE Serial monitor for that or do you recommend another program?

Thanks for the support.

**Comment by maragelis** [comment]
Trace=1 yes,  not TRACE=1 yes you can use Arduino IDE just choose right port

**Comment by maragelis** [comment]
If you have patience we will fix it I just need your help.

**Comment by rjduraocosta** [comment]
Perfect! You can count on me.

Tomorrow I will share the Trace output's with you.

Thanks again for all for the support.

**Comment by rjduraocosta** [comment]
Hello @maragelis,

I have the data for you. I restarted the system this morning just to start from scratch and then enabled the Trace and monitored the MQTT messages. For about 2 hours it was all stable with the reporting the zone events to the MQTT broker. After that period I stopped receiving the zone events and hads to send a "wake up" to the system by a command through MQTT, just like I described yesterday.
Attached you will find the Trace file. If you open it with and editor that you can see line numbers look at line 1138 that is when I "woke" the system up.

Thanks again for all your effort and help.
[MG5000_Trace_20190207.txt](https://github.com/maragelis/ParadoxRs232toMqtt/files/2840509/MG5000_Trace_20190207.txt)

**Comment by maragelis** [comment]
Ok thanks @rjduraocosta , I saw the problem. The serial buffer is getting corrupt. I do have code which tries to fix this, but looks like its getting stuck somewhere. I will try send you an update to see if this solves the problem. It should be ready by tonight be patient as I have day job too.

**Comment by rjduraocosta** [comment]
@maragelis, thank you for your work. Please take your time and when you are ready I will test it for you, no problem. I also have a day job and I know how thinks are no problem at all. I just had todays day off, remaining vacation day from last year, that is why I could test this today.

Thanks again and please take your time.

**Comment by maragelis** [comment]
I have a new cleaned up version 2.2 for testing and I thing I have squashed this bug. I also had this long time ago. It happens when to-many events overflow the serial buffer.  In the login i clear the buffer thats why it starts working again. Looks like its taking to long to process the incoming messages.

**Comment by maragelis** [comment]
Can you please try this version

https://github.com/maragelis/ParadoxRs232toMqtt/tree/Dev/ParadoxAlarmSystem/ParadoxAlarmSystemOTA

**Comment by rjduraocosta** [comment]
Yes of course, thank you.

Will give it a test and keep you posted on the results.

Thanks again.

**Comment by ciscozine** [comment]
Hi @maragelis, today after 3 days I have the same problem.
My cable is about 40cm. Could it be the cause of the disconnection?

thanks
Fabio

**Comment by maragelis** [comment]
No its a buffer issue

**Comment by ciscozine** [comment]
Ok, so it is possible fix via software?
thanks again 👍

**Comment by rjduraocosta** [comment]
Hello @maragelis,

I have been testing the last version last night in the evening. I noticed that the system stopped sending the zone events and I tried to send a command to "wake" the system up but this time I had the following  in the MQTT brooker:

![image](https://user-images.githubusercontent.com/40459138/52475218-c9a89900-2b92-11e9-85df-df1658506a9c.png)

Any ideas that I can test?

Thanks again

**Comment by maragelis** [comment]
give me 10 min

**Comment by maragelis** [comment]
I have just updated the dev project try that one https://github.com/maragelis/ParadoxRs232toMqtt/tree/Dev/ParadoxAlarmSystem/ParadoxAlarmSystemOTA

**Comment by ciscozine** [comment]
@maragelis Does the new release fix the buffer issue?

Thanks

**Comment by maragelis** [comment]
I myself HOPE SO

**Comment by ciscozine** [comment]
I'll load the new bin and I'll let you know  :)

**Comment by rjduraocosta** [comment]
> I have just updated the dev project try that one https://github.com/maragelis/ParadoxRs232toMqtt/tree/Dev/ParadoxAlarmSystem/ParadoxAlarmSystemOTA

Perfect will do so. I can only do it at the end of this afternoon. I will keep you posted on the results.

Thanks again.

**Comment by maragelis** [comment]
Great that gives me time to do some house keeping

**Comment by ciscozine** [comment]
It doesn't work.
I see:
paradoxdCTL/status - {"status":"Paradox connected"}
paradoxdCTL/status - SYSTEM READY PARADOX_2.2.0 
paradoxdCTL/status/Arm - 

The client is reachable, but it doesn't send any mqtt message to the broker.

**Comment by maragelis** [comment]
You will have to wait untill I get home. So I can check.

**Comment by maragelis** [comment]
I dont see a error on my side everything works fine

**Comment by ciscozine** [comment]
Do you see mqtt message to the broker?

**Comment by maragelis** [comment]
yes

**Comment by maragelis** [comment]
if you are compiling yourself change the PubSubClient.h file 
change 
#define MQTT_MAX_PACKET_SIZE 512

**Comment by maragelis** [comment]
How about Trace

**Comment by ciscozine** [comment]
> 
> 
> if you are compiling yourself change the PubSubClient.h file
> change
> #define MQTT_MAX_PACKET_SIZE 512

No good news. No mqtt message

**Comment by ciscozine** [comment]
If I try the bin file, I cannot see the hotspot to change the ip address of the mqtt broker

**Comment by ciscozine** [comment]
It seems that the

state_topic: "paradoxdCTL/out/state"

have been changed, because HASSIO cannot find the alarm state. Is it possible?

**Comment by maragelis** [comment]
can you please use mqtt.fx leave Hassio for the moment.

**Comment by ciscozine** [comment]
Yes sure,
do you need some output?

thanks

**Comment by maragelis** [comment]
I just updated. try the new one I did an update for you to see if that fixes it

**Comment by maragelis** [comment]
do yo have hangouts

**Comment by maragelis** [comment]
yes maragelis@gmail.com

**Comment by rjduraocosta** [comment]
Hello @maragelis , @ciscozine good evening,

Seems like I missed all the fun :). Where are we at the moment with this subject? Can I help and do some more tests? What should I test?

Thank again for your support.

**Comment by maragelis** [comment]
https://github.com/maragelis/ParadoxRs232toMqtt/tree/test/ParadoxAlarmSystem/ParadoxAlarmSystemOTA

**Comment by rjduraocosta** [comment]
Hello @maragelis ,

I am trying to test the above version but there are no MQTT messages coming. I am compiling myself with the suggested Arduino IDE settings, ESP8266 core 2.4.2 and #define MQTT_MAX_PACKET_SIZE 512 but nothing happens.
In the ino file I have the following:

#define Hassio 1
#define HomeKit 0
#define SendAllE0events 1 
#define SendEventDescriptions 0

bool TRACE = 0;
bool OTAUpdate = 0;

Is there something else different now?

Thanks for your help and support.

**Comment by maragelis** [comment]
When did you download the file . I you just downloaded 
These are the new topics 

"paradoxdCTL/out";
"paradoxdCTL/status";
"paradoxdCTL/in";
"paradoxdCTL/hassio/Arm";
"paradoxdCTL/hassio";
"paradoxdCTL/HomeKit";

**Comment by maragelis** [comment]
You mqtt.fx on topic paradoxdCTL/#

**Comment by maragelis** [comment]
Second when flashing new version **under tools** set **ERASE FLASH: ALL FLASH CONTENTS**

**Comment by rjduraocosta** [comment]
I downloaded them yesterday evening when you sent the link to test branch. I noticed that the topics changed a bit that is why I am subscribing to the paradoxdCTL/# in MQTT.fx

Maybe I will try to compile everything and do a another download to the wemos.

Thanks again.

**Comment by maragelis** [comment]
use the Stable V2.2.0 version

**Comment by rjduraocosta** [comment]
Yes that is the one I am using, I downloaded it from the test branch.

I am compiling everything again and will download afterwards to see.

Thanks.

**Comment by rjduraocosta** [comment]
Strange situation...Nothing happens. The AP does not even show up. I am flashing a blank_4MB file and the download everything again.

Will keep you posted.

**Comment by maragelis** [comment]
**under tools** set **ERASE FLASH: ALL FLASH CONTENTS**

**Comment by maragelis** [comment]
bring down #define MQTT_MAX_PACKET_SIZE 512 to 256

**Comment by rjduraocosta** [comment]
Thanks will do that.

**Comment by maragelis** [comment]
no need for blank_4MB if you set ERASE FLASH: ALL FLASH CONTENTS

**Comment by rjduraocosta** [comment]
I am using that option, but as I saw that nothing was happening I decided to try the blank, just in case.

**Comment by rjduraocosta** [comment]
Finally I am receiving zone events now. 

Just one question please. The Arm status is brought up at startup of the board in this version or just when requested? If so do we have to set up any option in the ion file?

Thank you again for your help.

**Comment by maragelis** [comment]
Just when requested (armstate command) , I tried to do it but without logging in (password) it cannot work.

**Comment by rjduraocosta** [comment]
Makes sence. You have also to enable this option #define HomeKit 1, to the Arm state to be reported.

System is up and running.

Thanks again for all your help.

**Comment by maragelis** [comment]
@rjduraocosta  could you please help @ciscozine  I dont have time.

**Comment by rjduraocosta** [comment]
Sure no problem. I have +/- 2/3 hours before I leave so @ciscozine post your doubts please. If it something of my knowledge I will help you.

**Comment by ciscozine** [comment]
Hi, now it works, but How can I check the alarm status?

**Comment by ciscozine** [comment]
@rjduraocosta can you ping the esp8266 and tell me what latency have you got? Thanks

**Comment by rjduraocosta** [comment]
> Hi, now it works, but How can I check the alarm status?

When you send a command (stay, arm, disarm, etc.) the status will updated in the topics

paradoxdCTL/Hassio/Arm
paradosdCTL/HomeKit

Or you can send an armstate comand also: { "password":"1234", "Command":"armstate", "Subcommand":"0" }






> @rjduraocosta can you ping the esp8266 and tell me what latency have you got? Thanks
 
Here you go:

![image](https://user-images.githubusercontent.com/40459138/52521510-06968d80-2c70-11e9-8da1-b9e7897812a5.png)

**Comment by ciscozine** [comment]
@rjduraocosta have you noticed this high latency also in the previous version?

**Comment by ciscozine** [comment]
This is mine:
Risposta da 10.255.200.127: byte=32 durata=190ms TTL=254
Risposta da 10.255.200.127: byte=32 durata=3ms TTL=254
Risposta da 10.255.200.127: byte=32 durata=22ms TTL=254
Risposta da 10.255.200.127: byte=32 durata=39ms TTL=254
Risposta da 10.255.200.127: byte=32 durata=58ms TTL=254
Risposta da 10.255.200.127: byte=32 durata=77ms TTL=254
Risposta da 10.255.200.127: byte=32 durata=88ms TTL=254
Risposta da 10.255.200.127: byte=32 durata=107ms TTL=254
Risposta da 10.255.200.127: byte=32 durata=24ms TTL=254
Risposta da 10.255.200.127: byte=32 durata=40ms TTL=254

in the previous version I have ~2ms

**Comment by rjduraocosta** [comment]
As I remember it has always been more or less the same. Do you think this is an issue for something you are experiencing?

**Comment by ciscozine** [comment]
well, I have the accesspoint 30cm from the esp8266. It is not normal has 100ms of latency.

**Comment by ciscozine** [comment]
@rjduraocosta do yoy have magellan 5000? do you have the same problem with this release? I have again the same problem.

**Comment by rjduraocosta** [comment]
I have another esp8266 in the same range, in fact in the same room and the latency is like this:

![image](https://user-images.githubusercontent.com/40459138/52521715-0ea3fc80-2c73-11e9-8965-c3b9c38ac0b1.png)


> @rjduraocosta do yoy have magellan 5000? do you have the same problem with this release? I have again the same problem.

Yes I have MG5000. I noticed the same issue, and when I see the system stopped I try and send a disarm, arm, etc. command and the Wemos responds to MQTT with a reconnection and then the system works again.

**Comment by ciscozine** [comment]
@rjduraocosta so it is not fixed the problem.

**Comment by rjduraocosta** [comment]
From the testing I performed today I am afraid you are right. 

@maragelis when you have time please tell me if I can test something else. Maybe another approach on something?

Thanks.

**Comment by maragelis** [comment]
Someone will have to send a trace file to me. Up untilll it stops working.

**Comment by maragelis** [comment]
I have one more version for you to try. Uploading in 10

**Comment by rjduraocosta** [comment]
Perfect. I will put to test as soon as you release it and do Trace also.

Thanks

**Comment by maragelis** [comment]
Lets try PubSubClient.h at 512 for this version 

https://github.com/maragelis/ParadoxRs232toMqtt/tree/Dev/ParadoxAlarmSystem/ParadoxAlarmSystemOTA

**Comment by maragelis** [comment]
after this I have one more trick and then I give up.

**Comment by pilievwm** [comment]
Hello @maragelis if i may, I would like to join the club of "Problem connecting to panel" :)
The wired thing is that I am receiving all Zone Events without any issue BUT I can not control my SP65 (actually I managed to arm it only once) and after that no luck! 
I am with the master, installed 4 hours ago.

**Comment by ciscozine** [comment]
@rjduraocosta Hi, with the last version I cannot receive the alarm state. Have you the same problem?
thanks

**Comment by pilievwm** [comment]
I can not build the latest version: 
`C:\Users\Peter\Documents\TEMP\ParadoxAlarmSystemOTA\ParadoxAlarmSystemOTA.ino: In lambda function:

ParadoxAlarmSystemOTA:184: error: 'getpage' was not declared in this scope

       HTTP.send(200, "text/html", getpage());

                                           ^

C:\Users\Peter\Documents\TEMP\ParadoxAlarmSystemOTA\ParadoxAlarmSystemOTA.ino: In function 'void processMessage(byte, byte, byte, String)':

ParadoxAlarmSystemOTA:388: error: 'getSubEvent' was not declared in this scope

       root["sub_eventD"]=getSubEvent(event,sub_event);

                                                     ^

ParadoxAlarmSystemOTA:389: error: 'getEvent' was not declared in this scope

       root["eventD"]=getEvent(event);

                                    ^

exit status 1
'getpage' was not declared in this scope`

**Comment by ciscozine** [comment]
Try to use .bin file. It is more simple than compiling

**Comment by rjduraocosta** [comment]
> Lets try PubSubClient.h at 512 for this version
> 
> https://github.com/maragelis/ParadoxRs232toMqtt/tree/Dev/ParadoxAlarmSystem/ParadoxAlarmSystemOTA

Sorry guys but I had a power loss in my area.

I am building the above version and test in a few minutes.

**Comment by pilievwm** [comment]
I test it.  I subscribe to paradoxdCTL/# and after few events i got 

`{"status":"Paradox Disconnected"}`

**Comment by ciscozine** [comment]
try to ping the esp8266. If you receive "Paradox Disconnected" you could have wireless problem

**Comment by rjduraocosta** [comment]
For the last version compiled with PubSubClient.h at 512 I am receiving live events from the panel. Even the controls work fine.
Now I really have to go out. I will leave the system testing. Tomorrow I will do a more thorough test with a Trace.

Thanks.

**Comment by pilievwm** [comment]
@rjduraocosta please tell me where do you define 512

**Comment by rjduraocosta** [comment]
Serch for PubSubClient in the windows search bar. Enter that folder and the the src folder. Open  PubSubClient.h with notepad++, for example, and search for the line #define MQTT_MAX_PACKET_SIZE and change the value to 512. Afterwards save and open the Arduino IDE and chose another board in the for it to fail compilation and then chose the correct board and compile again.

Hope I helped.

**Comment by pilievwm** [comment]
> try to ping the esp8266. If you receive "Paradox Disconnected" you could have wireless problem

The ping is alive. The problem is with the MQTT. It constantly gets disconnected with a message:
Client 80:7D:3A:5D:31:91 has exceeded timeout, disconnecting. 1549731148: Socket error on client 80:7D:3A:5D:31:91, disconnecting.

**Comment by pilievwm** [comment]
> #define MQTT_MAX_PACKET_SIZE

Awesome! Thanks for that help @rjduraocosta

**Comment by ciscozine** [comment]
hi @maragelis , after several test I get the same problem. I do not receive the mqtt message to the broker anymore

**Comment by pilievwm** [comment]
I am preaty sure that the problem is software. I tested with 3 wemos and none of them succeed to connect to the broker.

**Comment by maragelis** [comment]
Can you guys try the test branch version.

**Comment by pilievwm** [comment]
of course but which one?

**Comment by maragelis** [comment]
Change the branch to test and get the files

**Comment by maragelis** [comment]
Yes that one

**Comment by pilievwm** [comment]
Same result...

Working for few packages and stoped.
`Reply from 192.168.1.129: bytes=32 time=6ms TTL=128
Reply from 192.168.1.129: bytes=32 time=28ms TTL=128
Reply from 192.168.1.129: bytes=32 time=9ms TTL=128
Reply from 192.168.1.129: bytes=32 time=10ms TTL=128
Request timed out.
Reply from 192.168.1.129: bytes=32 time=9ms TTL=128
Reply from 192.168.1.129: bytes=32 time=11ms TTL=128
Reply from 192.168.1.129: bytes=32 time=3ms TTL=128
Reply from 192.168.1.129: bytes=32 time=32ms TTL=128
Reply from 192.168.1.129: bytes=32 time=33ms TTL=128
Reply from 192.168.1.129: bytes=32 time=6ms TTL=128
Reply from 192.168.1.129: bytes=32 time=8ms TTL=128
Reply from 192.168.1.129: bytes=32 time=9ms TTL=128
Reply from 192.168.1.129: bytes=32 time=31ms TTL=128
Reply from 192.168.1.129: bytes=32 time=2ms TTL=128
Reply from 192.168.1.129: bytes=32 time=14ms TTL=128
Reply from 192.168.1.129: bytes=32 time=15ms TTL=128
Reply from 192.168.1.129: bytes=32 time=18ms TTL=128
Request timed out.
Request timed out.
Request timed out.
Request timed out.
Request timed out.
Reply from 192.168.1.100: Destination host unreachable.
Reply from 192.168.1.100: Destination host unreachable.`

Same time I flush two other tasmota on wemos and their ping is flowless

**Comment by pilievwm** [comment]
Is there a way somehow to set static IP address for that device?

**Comment by pilievwm** [comment]
Only the master branch is working without a problem with WiFi connectivity. 
Tested!

**Comment by maragelis** [comment]
What panel are you using

**Comment by pilievwm** [comment]
Some how it is working now with the Master branch!
I can control the panel and it is not disconnecting.
So far so good 👍 

Thanks @maragelis for your continues work on that project!

Now i have 2 questions:

1. How to set a static IP address of the device?
2. Can you add retain on the alarm states so when the system is restarted to know the last state of the alarm?

**Comment by rjduraocosta** [comment]
Hello to all and good morning,

I left the latest Dev brach to the test yesterday evening and today morning when I checked the system was down and the Wemos was not responding to ping commands.

@maragelis I am going to start up the system again and enable the Trace to send it to you.

Thanks for all the support.

**Comment by maragelis** [comment]
@pilievwm can you please tell me how you installed the master did you compile ? With what settings? What core ? Or did you use the release bin file.

**Comment by maragelis** [comment]
I will use master as base and work up the changes. I have too strip most of the nice to have stuff as it looks as if the wemos cannot cope with the high serial demand.

**Comment by rjduraocosta** [comment]
It is not coming anything usefull on Serial output with Trace enable, just some odd characters.

@maragelis I am going to return to master, maybe compiled with a lower version of Arduino core like 2.3.0 helps? Please feel free to post the  updates that, with the time that I have, I will help you test them.

Thanks

**Comment by maragelis** [comment]
The dev version users hardware serial d4 for trace so you have to use a fridge adapter to se out put

**Comment by rjduraocosta** [comment]
> The dev version users hardware serial d4 for trace so you have to use a fridge adapter to se out put

Yes I am doing that, like I already did in previous dev versions, but the output is a bunch of odd characters that are not understandable.

Thanks.

**Comment by maragelis** [comment]
Set baud rate 115200

**Comment by ciscozine** [comment]
Hi @maragelis , is there some news about the anomaly in the software?
tell me if you need some test to help you.

what I find during the problem is that the mqtt socket is still live, but no mqtt message is sent.

thanks again

**Comment by pilievwm** [comment]
> @pilievwm can you please tell me how you installed the master did you compile ? With what settings? What core ? Or did you use the release bin file.

I did the installation with compiling it with these settings:
![image](https://user-images.githubusercontent.com/5072036/52534620-9447ab00-2d4c-11e9-8840-e82d56ae1c0a.png)

Everything is stable and working fine

**Comment by rjduraocosta** [comment]
> > @pilievwm can you please tell me how you installed the master did you compile ? With what settings? What core ? Or did you use the release bin file.
> 
> I did the installation with compiling it with these settings:
> ![image](https://user-images.githubusercontent.com/5072036/52534620-9447ab00-2d4c-11e9-8840-e82d56ae1c0a.png)
> 
> Everything is stable and working fine

@pilievwm wich Anrduino core are you using 2.4.2:

![image](https://user-images.githubusercontent.com/40459138/52535264-0798ef00-2d44-11e9-8b12-9d49a5561e8e.png)

This option was automatic or did you change it?

![image](https://user-images.githubusercontent.com/40459138/52535283-2e572580-2d44-11e9-9ea5-1180415fafb6.png)

Thanks.

**Comment by ciscozine** [comment]
@maragelis I have analyzed your 'test' code and I notice:
1. to receive the message "armed_way" from paradoxdCTL/hassio/Arm I modified the code
  `if   ((Hassio || HomeKit) && (event == 30 || (event==2 && sub_event==11)))`
with the code
 `if   ((Hassio || HomeKit) && ((event == 30 || (event==2 && sub_event==11)) || (event == 29 || (event==2 && sub_event==12) )  ))`


2. I think there an error in the if/else cycle:
I changed the code:

```
     else if ( paradox.sub_event == 4)
     {
       datachanged=true;
       hassioStatus.stringArmStatus = "armed_home";
       homekitStatus.stringArmStatus = "NIGHT_ARM";
       homekitStatus.intArmStatus = 2;
     }
   }
```

with the code

```
    else if ( paradox.sub_event == 4)
    {
      datachanged=true;
      hassioStatus.stringArmStatus = "armed_home";
      homekitStatus.stringArmStatus = "NIGHT_ARM";
      homekitStatus.intArmStatus = 2;
      
    }
    else{}
  }
  else{}

```

tell me what do you think

thanks.

**Comment by pilievwm** [comment]
@rjduraocosta My settings are the same except the LwPP which is 1.4 High Bandwidth 
The ESP board version is 2.4.2 same as yours

**Comment by maragelis** [comment]
@ciscozine not its not correct.  you dont need both else{} statements, and adding the too events to the if statement causes problems with smart home rules. Let me explain. 

When the alarm is armed for sleep, the alarm will also produce a full arm event and then the sleep event, this if you have a rule to run automation on sleep and other on arm the sleep arm event will run both.

@pilievwm how is the master branch going. I think I found the problem, whats causing the timeouts, Its the NTP procesure. The wemos is trying to get hold of ntp time and its timing out causing network errors. I will try fix tonight and get an update to you all. Stay tuned.

**Comment by rjduraocosta** [comment]
> @ciscozine not its not correct. you dont need both else{} statements, and adding the too events to the if statement causes problems with smart home rules. Let me explain.
> 
> When the alarm is armed for sleep, the alarm will also produce a full arm event and then the sleep event, this if you have a rule to run automation on sleep and other on arm the sleep arm event will run both.
> 
> @pilievwm how is the master branch going. I think I found the problem, whats causing the timeouts, Its the NTP procesure. The wemos is trying to get hold of ntp time and its timing out causing network errors. I will try fix tonight and get an update to you all. Stay tuned.

Hello @maragelis ,

To kkep you up to date with my testing, yesterday after some test with the latest Dev version and network timouts I returned to master and left the system to the test. I am still facing the same issue with the serial packages not beeing pocessed after some time. Afterwards I do the same procedure as before, I wake up the system with a command through MQTT, and all is up and running again.

Thank you for your dedication and support.

**Comment by ciscozine** [comment]
@maragelis 
Ok thank you.
Regarding the problem, it seems that the wemos start to be unresponsive during alarm/disarm command. Can be a serial buffer problem? Thanks

**Comment by maragelis** [comment]
201902111500 is up under dev branch if someone is willing to test.

**Comment by ciscozine** [comment]
I'll test this night :)
Thank you!

**Comment by rjduraocosta** [comment]
@maragelis, how to set the PubSub Mqtt max packet size for this dev version? 512?

Thanks

**Comment by maragelis** [comment]
i suggest 256

**Comment by rjduraocosta** [comment]
Hello again @maragelis ,

I had some time this afternoon, so I decided to put your new Dev version to the test. I did a Trace with this version. The live events are also stopping after some time, please find attached the Trace file and in lines 173 and 293 of the file I put some comments for you to understand what actions did I take.

Thanks again.
[Paradox_Trace_Dev_20190211.txt](https://github.com/maragelis/ParadoxRs232toMqtt/files/2851687/Paradox_Trace_Dev_20190211.txt)

**Comment by maragelis** [comment]
@rjduraocosta the trace was very helpfull thanks you. I would like you to try something. Looks like the chip has some problem with its spiffs. This has been noticed with some Chinese chips(puya memory)
Could you please try core 2.3.0 with wifiManager 0.12.0 and get back to me..with another trace. 
I wish I could reproduce this on my setup.

**Comment by pilievwm** [comment]
> @ciscozine not its not correct. you dont need both else{} statements, and adding the too events to the if statement causes problems with smart home rules. Let me explain.
> 
> When the alarm is armed for sleep, the alarm will also produce a full arm event and then the sleep event, this if you have a rule to run automation on sleep and other on arm the sleep arm event will run both.
> 
> @pilievwm how is the master branch going. I think I found the problem, whats causing the timeouts, Its the NTP procesure. The wemos is trying to get hold of ntp time and its timing out causing network errors. I will try fix tonight and get an update to you all. Stay tuned.

@maragelis it is working perfectly ok! There are no issues at all!
I just struggle with the retain messages because when the Home assistant is restarted the alarm is with unknown status and I have to send a manual MQTT message to arm and then to disarm it. Can you add a retain message to the system statuses?
And the other request is to have MQTT auth - I have other devices which use a cloud MQTT service. So my system needs to be accessible from the outside. Leaving the MQTT broker without authentication is huge security hole :)
And the last - I need my wemos to come to a static IP address. I really hate DHCP. Sometimes it gets a mess, the alarm is something really important and it needs a stable connection.

**Comment by maragelis** [comment]
which version is working for you

**Comment by pilievwm** [comment]
This one:
https://github.com/maragelis/ParadoxRs232toMqtt/blob/master/ParadoxAlarmSystem/ParadoxAlarmSystemOTA/ParadoxAlarmSystemOTA.ino

**Comment by maragelis** [comment]
@pilievwm in 201902112130 i added mqtt authentication and retain on hassio arm topic. If you want get the code and update master or try this dev version

**Comment by maragelis** [comment]
@rjduraocosta managed to reproduce the error on my test panel. I think I solved the problem. 
this is for @ciscozine  and @rjduraocosta  guys thanks for your help. 
@ciscozine  I need a full event subevent list when you arm please.

**Comment by rjduraocosta** [comment]
> @rjduraocosta managed to reproduce the error on my test panel. I think I solved the problem.
> this is for @ciscozine and @rjduraocosta guys thanks for your help.
> @ciscozine I need a full event subevent list when you arm please.

Fantástic @maragelis , I am glad you found it. I was about to send you another Traced with the Dev version compiled in version 2.3.0 and WifiManager 0.12.0, with the same output as before.

If you need anything please keep me posted. I will test the new releases if you need.

Thanks for your awsome work.

**Comment by maragelis** [comment]
Yes please test

**Comment by rjduraocosta** [comment]
[Paradox_Dev_Trace_20190211_Esp8266_2_3_0_WifiManager_0_12_0.txt](https://github.com/maragelis/ParadoxRs232toMqtt/files/2852789/Paradox_Dev_Trace_20190211_Esp8266_2_3_0_WifiManager_0_12_0.txt)
I already did. Attached there is the Trace that I mentioned above. The output is the same as the last one,

**Comment by maragelis** [comment]
@rjduraocosta are you using the bin file or compile

**Comment by maragelis** [comment]
looks like you are not using the right file 201902112130
https://github.com/maragelis/ParadoxRs232toMqtt/tree/Dev/ParadoxAlarmSystem/ParadoxAlarmSystemOTA

**Comment by rjduraocosta** [comment]
Oh sorry I did not know you already did the upload...my mistake...sorry.

I will test it tonight. Wich settings?

Esp8266 version 2.3.0, 2.4.2?
WifiManager version 0.12.0, 0.14.0?
MQTT max packet size?

Thanks for your support.

**Comment by maragelis** [comment]
I used 2.5.0
wifiManager 0.14.0
Mqtt 256

**Comment by rjduraocosta** [comment]
I will do the same with a Trace and report back.

Thanks.

**Comment by rjduraocosta** [comment]
Which option did you use for the Flash size?

![image](https://user-images.githubusercontent.com/40459138/52590743-00eca380-2e3a-11e9-9273-412b112e2790.png)

Thanks.

**Comment by maragelis** [comment]
That one the first

**Comment by maragelis** [comment]
Keep default

**Comment by ciscozine** [comment]
Hi, I tested the last dev version, but it does not connect to the mqtt broker (hassio).
Any suggestions?

```
1549920493: New connection from 10.255.200.127 on port 1883.
1549920493: Socket error on client <unknown>, disconnecting.
1549920498: New connection from 10.255.200.127 on port 1883.
1549920498: Socket error on client <unknown>, disconnecting.
1549920503: New connection from 10.255.200.127 on port 1883.
1549920503: Socket error on client <unknown>, disconnecting.
1549920508: New connection from 10.255.200.127 on port 1883.
1549920508: Socket error on client <unknown>, disconnecting.
```

I used 2.5.0
wifiManager 0.14.0
Mqtt 256


Thanks!
Fabio

**Comment by ciscozine** [comment]
UPDATE: Using mqtt + AUTH it works.
I'll test and let you know good news :)

**Comment by rjduraocosta** [comment]
@maragelis System is up and running with Trace enabled. As soon as I have some news I will keep you posted.

@ciscozine if you do not use atentication in your MQTT brooker, just use something like this in the ino file:
#define mqtt_user         "user"
#define mqtt_password     "pass"

**Comment by rjduraocosta** [comment]
Hello and good morning to all,

@maragelis you did it, it works! Thanks a lot for your dedication. Attached I am sending you a Trace of the test and as you can see at line 951 is where your new code solved the issue. Afterwards events where coming like they used to with no need to "wake up" the system. 

Perfect!

Another thing just for your info. In this version the data of the MQTT payload {"event":1,"sub_event":3,"data":""} is coming empty because the function   processMessage is beeing called with the last parameter with an empty string:

processMessage( inData[7], inData[8], "");

Great work and superb support.

When you need further testing for a master release please let me know.

Thanks again for you time.

[Paradox_Dev_Trace_201902112130_Fixed.txt](https://github.com/maragelis/ParadoxRs232toMqtt/files/2854836/Paradox_Dev_Trace_201902112130_Fixed.txt)

**Comment by maragelis** [comment]
i was getting a lot of garbage on data string which at first I thought might be throwing the buffer error, so i disabled it, I will renable it, and send the dev branch to TEST

**Comment by maragelis** [comment]
By the way @rjduraocosta without your help I would not of found it. Thank you.

**Comment by rjduraocosta** [comment]
@maragelis you are absolutely welcome. I am very glad that I could contribute.
You managed to turn our Alarm System in an interactive system with our Home Automation server (in my case openHAB). That is priceless work. 

Thnaks again for your time.

**Comment by rjduraocosta** [comment]
@maragelis new Dev branch version is up and runnig with Trace enbled.

I have to go out now and I will only come back late in the evening. I will keep you posted on the results and send you the Trace file.

Thanks again.

**Comment by rjduraocosta** [comment]
> @maragelis new Dev branch version is up and runnig with Trace enbled.
> 
> I have to go out now and I will only come back late in the evening. I will keep you posted on the results and send you the Trace file.
> 
> Thanks again.

@maragelis, just to update you on the test I was performing when you added the data string to the MQTT payload. Everything runs as expected. You can see in the attached Trace that your code did it's magic in line 1530.

Thanks again for your work.
[Paradox_Dev_20190212_Stable_2.2.1.txt](https://github.com/maragelis/ParadoxRs232toMqtt/files/2857416/Paradox_Dev_20190212_Stable_2.2.1.txt)

**Comment by maragelis** [comment]
I think we can close the issue. Thanks to all
